tools/cmake/toolchain.cmake

```diff
@@ -48,18 +48,10 @@ set(CMAKE_CXX_COMPILER_WORKS TRUE)
 
 # Set the language standard to use per target
 function(mbed_set_language_standard target)
-    set_target_properties(${target}
-        PROPERTIES
-            C_STANDARD 11
-            C_STANDARD_REQUIRED YES
-            C_EXTENSIONS YES
-    )
-
-    set_target_properties(${target}
-        PROPERTIES
-            CXX_STANDARD 14
-            CXX_STANDARD_REQUIRED YES
-            CXX_EXTENSIONS YES
+    target_compile_features(${target}
```

Comment:
The idea from Rob's comment was to remove `mbed_set_language_standard()` and just use `target_compile_features()`. Not to have `mbed_set_language_standard()` as a wrapper.

Reply:
I see, that would work. 👍

Reply:
Making it not breaking:
1. Update all examples to not to use this function
2. remove the function from our CMake (separate PR)

Reply:
@0xc0170  I have tried locally with removed `mbed_configure_app_target` and kept target_compile_features in toolchain.cmake, it fails for me, is it working for you?

Reply:
> Making it not breaking:
> 
> 1. Update all examples to not to use this function
> 2. remove the function from our CMake (separate PR)

Applications use `mbed_configure_app_target()` which can perhaps remain as such in case we need to add more stuff in the future. `mbed_configure_app_target()` however must use `target_compile_features()` directly and not `mbed_set_language_standard()`. Therefore, there would be a need to update examples.